swhkd/src/daemon.rs

```diff
@@ -501,50 +495,20 @@ pub fn set_command_line_args() -> Command<'static> {
     app
 }
 
-pub fn fetch_xdg_config_path() -> PathBuf {
-    let config_file_path: PathBuf = match env::var("XDG_CONFIG_HOME") {
-        Ok(val) => {
-            log::debug!("XDG_CONFIG_HOME exists: {:#?}", val);
-            Path::new(&val).join("swhkd/swhkdrc")
-        }
-        Err(_) => {
-            log::error!("XDG_CONFIG_HOME has not been set.");
-            Path::new("/etc/swhkd/swhkdrc").to_path_buf()
-        }
-    };
-    config_file_path
+pub fn fetch_xdg_config_path(env: &Env) -> PathBuf {
```

Comment:
Now that we are separating the env parsing from the daemon code, I think it would be apt to keep all env fetch functions in the env impl.

Reply:
Yeah that makes sense! Gonna do that

Reply:
Thanks Ishan!